### ISSUE TITLE
Very low-resolution grids

### ISSUE THREAD

**Issue by GeorgeGayno-NOAA** [NOAA-EMC/UFS_UTILS#1000]
The UFS group want to run the model at very low resolution - C24, C18 and C12. 

Check if UFS_UTILS can create grids this coarse.

Add relevant unit tests.

Related issues: 
#1004
https://github.com/ufs-community/ufs-weather-model/issues/2508

**Comment by GeorgeGayno-NOAA** [comment]
Updated the `cpld_gridgen` program (91c08f4) to output C12, C18 and C24. The program ran to completion and a spot check of the files did not indicate any problems. 

The files were placed on Hera: `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/fix/orog.lowres`

@DeniseWorthen - can you take a quick look?

**Comment by DeniseWorthen** [comment]
@GeorgeGayno-NOAA It seems to have generated the expected values. 

I didn't check every combo, but I did some spot-checking. This for example is a "normal" c96.mx100 plot of land-frac

<img width="400" alt="c96 mx100_landfrac" src="https://github.com/user-attachments/assets/f15fa5a9-63c9-4ae5-ae84-9d48bd333829">

and this is C12.mx025

<img width="400" alt="c12 mx025_landfrac" src="https://github.com/user-attachments/assets/e1ed96d0-06a1-4144-959b-b12341b8d3fe">

The land masses seem consistent, so I agree the generated files appear OK.

**Comment by GeorgeGayno-NOAA** [comment]
The following grids were created and placed on Hera in `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/fix/orog.lowres`

- C12.mx900
- C18.mx500
- C24.mx500

**Comment by GeorgeGayno-NOAA** [comment]
When creating the unit test for "remove_isolated_pts", I discovered a possible problem at the i-direction edges of the grid. To determine if a point is 'isolated', the surrounding 8 points are checked. So, if point (2,2) is checked for a 3x3 grid, the 8 surrounding points are:

```
 check point            2           2
 check points to the w            1           2
 check points to the e             3           2
 check points to the nw          1           1
 check points to the n             2           1
 check points to the ne           3           1
 check points to the sw           1           3
 check points to the s             2           3
 check points to the se           3           3
```
That is correct.

When point (1,2) is checked, some of the eight points are located at i=3. 

```
 check point            1           2
 check points to the w            3           2
 check points to the e            2           2
 check points to the nw          3           1
 check points to the n            1           1
 check points to the ne           2           1
 check points to the sw           3           3
 check points to the s             1           3
 check points to the se           2           3
```

That is only correct for a global grid. I suspect when the `orog` program was converted to run on tiles (instead of the global GFS grid), this logic was not adjusted. Without modifying the `orog` program to process all tiles are once, there is no way to properly check if all edge points are 'isolated'.  For now, I will simply not process the 'i' edges. Note, the 'j' edges were not being processed, which makes sense for an algorithm used for the old global GFS grid.

This test was done using 25c6b54.

**Comment by GeorgeGayno-NOAA** [comment]
> When creating the unit test for "remove_isolated_pts", I discovered a possible problem at the i-direction edges of the grid. To determine if a point is 'isolated', the surrounding 8 points are checked. So, if point (2,2) is checked for a 3x3 grid, the 8 surrounding points are:
> 
> ```
>  check point            2           2
>  check points to the w            1           2
>  check points to the e             3           2
>  check points to the nw          1           1
>  check points to the n             2           1
>  check points to the ne           3           1
>  check points to the sw           1           3
>  check points to the s             2           3
>  check points to the se           3           3
> ```
> 
> That is correct.
> 
> When point (1,2) is checked, some of the eight points are located at i=3.
> 
> ```
>  check point            1           2
>  check points to the w            3           2
>  check points to the e            2           2
>  check points to the nw          3           1
>  check points to the n            1           1
>  check points to the ne           2           1
>  check points to the sw           3           3
>  check points to the s             1           3
>  check points to the se           2           3
> ```
> 
> That is only correct for a global grid. I suspect when the `orog` program was converted to run on tiles (instead of the global GFS grid), this logic was not adjusted. Without modifying the `orog` program to process all tiles are once, there is no way to properly check if all edge points are 'isolated'. For now, I will simply not process the 'i' edges. Note, the 'j' edges were not being processed, which makes sense for an algorithm used for the old global GFS grid.
> 
> This test was done using [25c6b54](https://github.com/ufs-community/UFS_UTILS/commit/25c6b54109474087769ec4b9b0e686314ee9c382).

This change to "remove_isolated_pts" was made at a7545ea. All `grid_gen` regression tests passed.

**Comment by GeorgeGayno-NOAA** [comment]
While writing the unit test for 'remove_isolated_pts', I could not figure out how to trip the 'ELSE' branch of this 'IF' block. 
https://github.com/ufs-community/UFS_UTILS/blob/21c51d7a32f207526d2267f4b85133ee46fe0334/sorc/orog_mask_tools.fd/orog.fd/orog_utils.F90#L602

My guess is that this logic was used for the old GFS gaussian reduced grid - where the number of points in the 'i' direction decreased toward the poles. I will remove this logic.

**Comment by GeorgeGayno-NOAA** [comment]
> While writing the unit test for 'remove_isolated_pts', I could not figure out how to trip the 'ELSE' branch of this 'IF' block.
> 
> https://github.com/ufs-community/UFS_UTILS/blob/21c51d7a32f207526d2267f4b85133ee46fe0334/sorc/orog_mask_tools.fd/orog.fd/orog_utils.F90#L602
> 
> My guess is that this logic was used for the old GFS gaussian reduced grid - where the number of points in the 'i' direction decreased toward the poles. I will remove this logic.

The routine was updated at 380a57c. All `grid_gen` regression tests passed.

**Comment by GeorgeGayno-NOAA** [comment]
Mid-point check: all `grid_gen` regression tests passed using 858629b on Hercules.

**Comment by GeorgeGayno-NOAA** [comment]
@DeniseWorthen - to get `cpld_gridgen` to run at C12/18/24, I hardwired some changes at 91c08f4. Do you want to update the code more robustly for these resolutions?

**Comment by DeniseWorthen** [comment]
Yes, Thanks. I'll update it to read a list of resolutions, now that the previous PR w/ the generalized resolutions is done.

**Comment by DeniseWorthen** [comment]
@GeorgeGayno-NOAA I have a draft PR (#1013) for generation of the new low-res configurations on the ocean/ice side. But until the mosaic files are present in the fix directories, I still have some settings manually pointing to my own copy of the 'low res' directory.

For cpld_gridgen, the rt.conf has been modified to accept a list of resolutions, if present. Otherwise all available ATM resolutions will be used.

**Comment by GeorgeGayno-NOAA** [comment]
> [@GeorgeGayno-NOAA](https://github.com/GeorgeGayno-NOAA) I have a draft PR ([#1013](https://github.com/ufs-community/UFS_UTILS/pull/1013)) for generation of the new low-res configurations on the ocean/ice side. But until the mosaic files are present in the fix directories, I still have some settings manually pointing to my own copy of the 'low res' directory.
> 
> For cpld_gridgen, the rt.conf has been modified to accept a list of resolutions, if present. Otherwise all available ATM resolutions will be used.

I will open an issue to host the 'grid' and 'mosaic' files in the official 'fixed' directories.
